pyboreas/data/himmelsbach.py

```diff
@@ -153,17 +155,16 @@ def compute_model_and_get_inliers(self):
                         line_set = []
                         c += 1
                 else:
-                    if (len(lines) > 0):
-                        dprev = self.distpointline(lines[c - 1], idx)
-                    else:
-                        dprev = -1
-                    if (dprev <= self.Tdprev or c == 0 or len(line_set) != 0):
+                    mprev = np.abs(lines[-1].m) if len(lines) > 0 else 0
+                    dprev = self.distpointline(lines[c - 1], idx) if len(lines) > 0 else -1
+                    if (mprev > self.Tm or dprev <= self.Tdprev or c == 0 or len(line_set) != 0):
```

Comment:
the extra `mprev > self.Tm` condition is not part of the original algorithm -  I think original algorithm assumes that points close to the sensor must all be on the ground, which is not necessarily true. adding this condition allows us to ignore smoothness constraint when the slope of the previous line is too large - seems to address the issues where a sector of points are not segmented correctly